engine/impl/Playlist.php

```diff
@@ -162,39 +164,86 @@ public function updatePlaylist($playlist, $date, $time, $description, $airname)
     }
 
     public function getTrack($id) {
-        $query = "SELECT tag, artist, track, album, label, id FROM tracks " .
+        $query = "SELECT created, tag, artist, track, album, label, id FROM tracks " .
                  "WHERE id = ?";
         $stmt = $this->prepare($query);
         $stmt->bindValue(1, (int)$id, \PDO::PARAM_INT);
         return $this->executeAndFetch($stmt);
     }
     
     public function getTracks($playlist, $desc = 0) {
-        $query = "SELECT tag, artist, track, album, label, id FROM tracks " .
+        $query = "SELECT created, tag, artist, track, album, label, id FROM tracks " .
                  "WHERE list = ? ORDER BY id";
         if($desc)
             $query .= " DESC";
         $stmt = $this->prepare($query);
         $stmt->bindValue(1, (int)$playlist, \PDO::PARAM_INT);
         return $this->execute($stmt);
     }
-    
-    public function insertTrack($playlist, $tag, $artist, $track, $album, $label) {
+
+    private static function logIt($msg) {
+        $out = fopen('php://stderr', 'w'); //output handler
+        fputs($out, $msg."\n"); //writing output operation
+        fclose($out);
+    }
+
+    // return true if "now" is within the show start/end time & date.
+    // NOTE: this routine must be tolerant of improperly formatted dates.
+    public function isWithinShow($listRow) {
+        // TODO: set local TZ on startup.
+        //$TZ_LOCAL  = new DateTimeZone(date_default_timezone_get());
+
+        $TZ_LOCAL  = new DateTimeZone("America/Los_Angeles");
+        $TIME_FORMAT = "Y-m-d Gi"; // eg, 2019-01-01 1234
+        $retVal = false;
+
+        try {
+            $timeAr = explode("-", $listRow[2]);
+            if (count($timeAr) == 2) {
+                $timeStr1 = $listRow[1] . " " . $timeAr[0];
+                $start = DateTime::createFromFormat($TIME_FORMAT, $timeStr1, $TZ_LOCAL);
+                $endStr = $timeAr[1] == "0000" ? "2359" : $timeAr[1];
+                $timeStr2 = $listRow[1] . " " . $endStr;
+                $end = DateTime::createFromFormat($TIME_FORMAT, $timeStr2, $TZ_LOCAL);
+
+                if (isset($start) && isset($end)) {
+                    $now = new DateTime("now", $TZ_LOCAL);
```

Comment:
Use DateTime() and DateTime::createFromFormat() without any timezone parameter.  This will use the PHP default 'date.timezone' value, which represents the local timezone.  I suggest to put in the README.md that 'date.timezone' needs to be set in php.ini.